mne_bids/tests/test_write.py

```diff
@@ -2127,6 +2127,123 @@ def test_undescribed_events(_bids_validate, drop_undescribed_events):
     _bids_validate(bids_root)
 
 
+@pytest.mark.parametrize(
+    'dir_name, fname, reader, datatype, coord_frame', [
+        ('EDF', 'test_reduced.edf', _read_raw_edf, 'ieeg', 'mni_tal'),
+        ('EDF', 'test_reduced.edf', _read_raw_edf, 'ieeg', 'fs_tal'),
+        ('EDF', 'test_reduced.edf', _read_raw_edf, 'ieeg', 'unknown'),
+        ('EDF', 'test_reduced.edf', _read_raw_edf, 'eeg', 'head'),
+        ('EDF', 'test_reduced.edf', _read_raw_edf, 'eeg', 'mri'),
+        ('EDF', 'test_reduced.edf', _read_raw_edf, 'eeg', 'unknown'),
+        ('CTF', 'testdata_ctf.ds', _read_raw_ctf, 'meg', ''),
+        ('MEG', 'sample/sample_audvis_trunc_raw.fif', _read_raw_fif, 'meg', ''),  # noqa
+    ]
+)
```

Comment:
@adam2392 I don't have time to review in details but please monitor test time when you add tests. Add such big parametrize has a cost in CI time and needs to be fully justified.  It's a general statement. I'd to avoid having mne-bids having a test time lasting tens of minutes. I find it already quite long actually

Reply:
I ran these locally and it's actually quite fast because all these files are very very small. The chunk using `write_raw_bids` as a result is very fast.

The tests that generally seem to be taking long periods of time are actually the ones where we test for `split` of fif files correctly (need a humongous chunk to split it up > 2GB?), and the ones where we test the inspector. This is due to qualitative observation though, not an explicit profiling.

Reply:
I can take these out, but I added these cases for the sake of covering a multiple coordsystems per datatype. Do we want to limit the parametrizations?

Reply:
Actually pretty easy cuz I have worked with pytest-profiler before:

```
Profiling (from /Users/adam2392/Documents/mne-bids/prof/combined.prof):
Thu Jan  7 16:19:56 2021    /Users/adam2392/Documents/mne-bids/prof/combined.prof

         100182265 function calls (96919567 primitive calls) in 218.726 seconds

   Ordered by: cumulative time
   List reduced from 6828 to 20 due to restriction <20>

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
      179    0.002    0.000  218.719    1.222 runner.py:106(pytest_runtest_protocol)
      179    0.002    0.000  218.650    1.222 runner.py:114(runtestprotocol)
      537    0.005    0.000  218.647    0.407 runner.py:212(call_and_report)
3506/1979    0.018    0.000  218.642    0.110 hooks.py:272(__call__)
3506/1979    0.003    0.000  218.627    0.110 manager.py:90(_hookexec)
3506/1979    0.008    0.000  218.625    0.110 manager.py:84(<lambda>)
3506/1979    0.046    0.000  218.619    0.110 callers.py:157(_multicall)
      537    0.003    0.000  216.582    0.403 runner.py:240(call_runtest_hook)
      537    0.006    0.000  216.573    0.403 runner.py:298(from_call)
      537    0.002    0.000  216.564    0.403 runner.py:255(<lambda>)
      179    0.001    0.000  215.183    1.202 runner.py:153(pytest_runtest_call)
      179    0.001    0.000  215.179    1.202 python.py:1639(runtest)
      179    0.272    0.002  215.169    1.202 python.py:176(pytest_pyfunc_call)
       56    0.001    0.000   81.284    1.451 conftest.py:29(_validate)
       56    0.007    0.000   81.283    1.451 <decorator-gen-2>:1(run_subprocess)
       56    0.073    0.001   81.274    1.451 misc.py:90(run_subprocess)
     7964    0.142    0.000   80.728    0.010 queue.py:153(get)
     7265    0.110    0.000   80.532    0.011 threading.py:270(wait)
    33516   80.368    0.002   80.368    0.002 {method 'acquire' of '_thread.lock' objects}
      203    0.053    0.000   42.583    0.210 write.py:904(write_raw_bids)


======================================================================== slowest 20 durations =========================================================================
35.23s call     mne_bids/tests/test_read.py::test_write_read_fif_split_file
19.52s call     mne_bids/tests/test_write.py::test_write_anat
8.63s call     mne_bids/tests/test_write.py::test_fif
6.81s call     mne_bids/tests/test_inspect.py::test_inspect_annotations
6.54s call     mne_bids/tests/test_write.py::test_eegieeg[Persyst-sub-pt1_ses-02_task-monitor_acq-ecog_run-01_clip2.lay-fn]
6.37s call     mne_bids/tests/test_inspect.py::test_inspect_set_and_unset_bads
6.27s call     mne_bids/tests/test_write.py::test_eegieeg[NihonKohden-MB0400FU.EEG-fn]
6.23s call     mne_bids/tests/test_inspect.py::test_inspect_annotations_remove_all
6.15s call     mne_bids/tests/test_write.py::test_eegieeg[EDF-test_reduced.edf-fn]
5.64s call     mne_bids/tests/test_write.py::test_set
5.56s call     mne_bids/tests/test_inspect.py::test_inspect_bads_and_annotations
3.99s call     mne_bids/tests/test_write.py::test_ctf
3.87s call     mne_bids/tests/test_update.py::test_update_sidecar_jsons
3.84s call     mne_bids/tests/test_inspect.py::test_inspect_single_file[True]
3.29s call     mne_bids/tests/test_inspect.py::test_inspect_single_file[False]
3.28s call     mne_bids/tests/test_write.py::test_mark_bad_channels[MEG 0112-Really bad!-True-True-existing_ch_names7-existing_descriptions7-None-False]
3.19s call     mne_bids/tests/test_write.py::test_mark_bad_channels[ch_names0-None-False-False-existing_ch_names0-existing_descriptions0-None-False]
3.19s call     mne_bids/tests/test_write.py::test_write_meg_calibration
3.18s call     mne_bids/tests/test_write.py::test_mark_bad_channels[ch_names8-descriptions8-False-False-existing_ch_names8-existing_descriptions8-None-False]
3.17s call     mne_bids/tests/test_write.py::test_mark_bad_channels[ch_names6-descriptions6-False-False-existing_ch_names6-existing_descriptions6-meg-False]
```

If anything, I think the `test_eegieeg` I parametrized could be widdled down (if possible). Otw, these other tests are at least for now the longest 20.